controllers/gitrepository_controller_test.go

```diff
@@ -645,10 +645,11 @@ func TestGitRepositoryReconciler_reconcileSource_checkoutStrategy(t *testing.T)
 				}
 				conditions.MarkTrue(obj, sourcev1.ArtifactInStorageCondition, meta.SucceededReason, "foo")
 			},
-			want:                 sreconcile.ResultEmpty,
-			wantErr:              true,
-			wantRevision:         "staging/<commit>",
-			wantArtifactOutdated: false,
+			want:                  sreconcile.ResultEmpty,
+			wantErr:               true,
+			wantRevision:          "staging/<commit>",
+			wantArtifactOutdated:  false,
+			skipForImplementation: "libgit2",
```

Comment:
What's the reason for skipping optimized clone for libgit2?
As other test cases with skip option set when the results are specific to one implementation and there's a second case for the other implementation, the reason is understandable. Commenting the reason in the code would be helpful.

Reply:
Optimized clones are only enabled when using the libgit2 managed transport, which the controller tests don't. When we get rid of all the unmanaged transport code, we can avoid skipping this test again. Adding a comment for future reference 👍